tests/service/XmlDocTests.fs

```diff
@@ -0,0 +1,350 @@
+﻿#if INTERACTIVE
+#r "../../artifacts/bin/fcs/net461/FSharp.Compiler.Service.dll" // note, build FSharp.Compiler.Service.Tests.fsproj to generate this, this DLL has a public API so can be used from F# Interactive
+#r "../../artifacts/bin/fcs/net461/nunit.framework.dll"
+#load "FsUnit.fs"
+#load "Common.fs"
+#else
+module Tests.XmlDoc
+#endif
+
+open FSharp.Compiler.Service.Tests.Common
+open FSharp.Compiler.Symbols
+open FsUnit
+open NUnit.Framework
+
+let checkXml symbolName docs checkResults =
+    let symbol = findSymbolByName symbolName checkResults
+    let xmlDoc =
+        match symbol with
+        | :? FSharpMemberOrFunctionOrValue as v -> v.XmlDoc
+        | :? FSharpEntity as t -> t.XmlDoc
+        | :? FSharpUnionCase as u -> u.XmlDoc
+        | _ -> failwith $"unexpected symbol type {symbol.GetType()}"
+    match xmlDoc with
+    | FSharpXmlDoc.FromXmlText t -> t.UnprocessedLines |> shouldEqual docs
+    | _ -> failwith "wrong XmlDoc kind"
+
+let checkXmls data checkResults =
+    for symbolName, docs in data do checkXml symbolName docs checkResults
+
+[<Test>]
+let ``simple type xml doc``() =
+    let _, checkResults = getParseAndCheckResults """
+///A
+type A = class end
+"""
+    checkResults
+    |> checkXml "A" [|"A"|]
+
+[<Test>]
+let ``simple signature type xml doc``() =
+    let _, checkResults = getParseAndCheckResultsOfSignatureFile """
+///A1
+///A2
+type
+    ///A3
+    internal
+             ///A4
+             A
+"""
+    checkResults
+    |> checkXml "A" [|"A1"; "A2"|]
+
+[<Test>]
+let ``xml doc before/after type name``() =
+    let _, checkResults = getParseAndCheckResults """
+///A1
+type
+    ///A2
+    [<NotNull>]
+    A = class end
+"""
+    checkResults
+    |> checkXml "A" [|"A1"|]
+
+[<Test>]
+let ``multiline type xml doc``() =
+    let _, checkResults = getParseAndCheckResults """
+///A
+
+///B
+type A = class end
+"""
+    checkResults
+    |> checkXml "A" [|"A"; "B"|]
+
+[<Test>]
+let ``separated type xml doc``() =
+    let _, checkResults = getParseAndCheckResults """
+///A
+()
+///B
+type A = class end
+"""
+    checkResults
+    |> checkXml "A" [|"B"|]
+
+[<Test>]
+let ``separated by simple comment type xml doc``() =
+    let _, checkResults = getParseAndCheckResults """
+///A
+// Simple comment delimiter
+///B
+type A = class end
+"""
+    checkResults
+    |> checkXml "A" [|"B"|]
+
+[<Test>]
+let ``separated by multiline comment type xml doc``() =
+    let _, checkResults = getParseAndCheckResults """
+///A
+(* Multiline comment
+delimiter *)
+///B
+type A = class end
+"""
+    checkResults
+    |> checkXml "A" [|"B"|]
+
+[<Test>]
+let ``separated by star type xml doc``() =
+    let _, checkResults = getParseAndCheckResults """
+///A
+(*)
+///B
+type A = class end
+"""
+    checkResults
+    |> checkXml "A" [|"B"|]
+
```

Comment:
Could you please also add a test case for attributes before an xml doc in a type declaration?

Reply:
It's already there!
https://github.com/dotnet/fsharp/pull/11973/files/3ad240c448a9b49502cb011eca39a7ece0dc7a9b#diff-6353bbdf824b76a75ac480672baa99c8fb36c0379ec4170959c479acb4954bf5R270

Reply:
Ah, sorry, I didn't spot it there, thanks!